addons/stock/models/product.py

```diff
@@ -597,37 +590,47 @@ def write(self, vals):
                 raise UserError(_("You can not change the type of a product that is currently reserved on a stock move. If you need to change the type, you should first unreserve the stock move."))
         return super(ProductTemplate, self).write(vals)
 
-    def action_update_quantity_on_hand(self):
-        default_product_id = self.env.context.get('default_product_id', self.product_variant_id.id)
-        if self.env.user.user_has_groups('stock.group_stock_multi_locations') or (self.env.user.user_has_groups('stock.group_production_lot') and self.tracking != 'none'):
-            product_ref_name = self.name + ' - ' + datetime.today().strftime('%m/%d/%y')
-            ctx = {'default_filter': 'product', 'default_product_id': default_product_id, 'default_name': product_ref_name}
+    def action_open_quants(self, products=None):
+        self.env['stock.quant']._quant_tasks()
+        if not products:
+            products = self.mapped('product_variant_ids')
+        location_domain = products._get_domain_locations()[0]
+        domain = expression.AND([[('product_id', 'in', products.ids)], location_domain])
+        context = {'hide_location': not self.user_has_groups('stock.group_stock_multi_locations')}
+
+        # If user have rights to write on quant, we display the editable quant view.
+        if self.user_has_groups('stock.group_stock_manager'):
+            # Set default location id if multilocations is inactive
+            if not self.user_has_groups('stock.group_stock_multi_locations'):
+                user_company = self.env.user.company_id
+                warehouse = self.env['stock.warehouse'].search(
+                    [('company_id', '=', user_company.id)], limit=1
+                )
+                if warehouse:
+                    context.update({'default_location_id': warehouse.lot_stock_id.id})
```

Comment:
You should not always define the default_location_id even if the user do not use multi locations?

Reply:
I don't think because `location_id` is required, and this field is visible in multilocation (but isn't if multilocation is disabled and if you came from product view, so it's important to set a default in this case).
So, user have to set it, and I think it's less a cause of error if user set it manually than if it set automatically and then maybe user pass over without pay attention.